schema.graphql

```diff
@@ -40,6 +43,52 @@ type Query {
     ): SlateLineup
 }
 
+type CorpusSlate @key(fields: "id") {
+    id: ID!
+    name: String!
+    description: String!
+    experiments: [CorpusSlateExperiment!]!
+    recommendationResult: CorpusSlateRecommendationResult
+}
+
+type CorpusSlateExperiment @key(fields: "id") {
+    id: ID!
+    corpusSlateId: ID!
+    corpusSlate: CorpusSlate!
+    description: String!
+    candidateSets: [ID!]
+    rankers: [ID!]
+}
+
+type CorpusSlateRecommendationResult {
+    corpusSlateExperimentId: ID! #TODO: This existed in old Slate. I dont know how it gets populated.
+    corpusSlateExperiment: CorpusSlateExperiment!
+
+    """
+    An ordered list of the recommendations to show to the user
+    """
+    recommendations(count: Int = 10): [CorpusItemRecommendation!]!
+}
+
+"""
+Represents a Recomendation from the curated corpus
+"""
+type CorpusItemRecommendation @key(fields: "id") {
+    """
+    A unique identifier for this specific recommendation across all recommendations. This identifier is generated by the
+    a machine learning system making recommendation decisions. This field can be joined with recommendation decisions to
+    aggregate more meta data about the decision.
+    """
+    id: ID!
+
+    """
+    Corpus item id.
+    """
+    corpusItemId: ID!
```

Comment:
Corpus management tool will have to extend CorpusItemRecommendation to include `corpusItem: CorpusItem!`.

Reply:
> Would it be better, is it possible, to define the corpusItem property here?

This should only define what it knows about which is the corpusItemId. Then we let the subgraphs that know what to do with that extend and implement it. 

The key thing to remember is clients see the built graph and this will not matter to them. 


Now.... So now that I say this, i then scroll down and see Recommendation implements Item... and somehow resolves.... so we could also do it that way and CuratedCorpus never needs to know about recomendation api. I guess it depends on how we want to implement our graphs.